Bottom: 2f80746fa1d4b183672a8dee5f539fa1def038e2
Top:    a436a7c6e63d808104953e50bcf6f9135af7b645
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-05-21 14:12:02 +0800

Refresh of virtio-net-reset-num_buf-to-1

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index d9637ec..c57d588 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -536,8 +536,6 @@ static struct page *xdp_linearize_page(struct receive_queue *rq,
 		put_page(p);
 	}
 
-	/* receive_mergeable() assumes there's at least one buffer */
-	*num_buf = 1;
 	/* Headroom does not contribute to packet length */
 	*len = page_off - VIRTIO_XDP_HEADROOM;
 	return page;
@@ -730,6 +728,7 @@ static struct sk_buff *receive_mergeable(struct net_device *dev,
 						      &len);
 			if (!xdp_page)
 				goto err_xdp;
+			num_buf = 1;
 			offset = VIRTIO_XDP_HEADROOM;
 		} else {
 			xdp_page = page;
